### PR TITLE
core: topology cache database schema

### PR DESCRIPTION
**Pull request by mcutalo88**
### Description

This lays the foundation for the topology API, we are first starting by implementing the [RFC 1 topology cache](https://docs.google.com/document/d/13AYDcTOF39QRoo7fkRSVPpeGqPEowa8WnDWrifBlxmc). This PR sets up the datastore to store cached items throughout ones infrastructure. In addition it adds basic configuration for the topology feature set with some settings specific to the cache's behavior. 

### Testing Performed
Testing on the schema was done in [db-fiddle](https://www.db-fiddle.com/f/8EzuUpwXp9pEPHKS7TnHs/18).

### GitHub Issue

Fixes #
This is the building blocks for https://github.com/lyft/clutch/issues/30.

### CODE REVIEW

File: backend/cmd/migrate/migrations/000005_create_topology_cache_table.up.sql [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  id varchar UNIQUE,
+  
+  -- id_text_search: This is the same value as the id but in tsquery type for better indexing and fuzzy lookups
+  id_text_search tsquery,
+  
+  -- data: json blob of resource
+  data JSONB,
+  
+  -- resolver_type: is the infrastructure type that the resource belongs to, this could be `kubernetes` or `aws`
+  resolver_type varchar,
```
Comment (danielhochman) [inline backend/cmd/migrate/migrations/000005_create_topology_cache_table.up.sql:27]: per our earlier discussion, on second thought we will need to add the full type information i'm pretty sure.

if i'm on the pod terminate page and i start type `clutc` i only want to get pod IDs not clusters, but currently it's not possible to restrict it to that, is it?

Comment (mcutalo88) [reply backend/cmd/migrate/migrations/000005_create_topology_cache_table.up.sql:27]: Yeah true will want to have the full resource proto type like the `typeUrl`, let me make some changes.

```
type.googleapis.com/clutch.k8s.v1.Deployment
type.googleapis.com/clutch.k8s.v1.Pod
type.googleapis.com/clutch.aws.ec2.v1.Instance
```

File: backend/cmd/migrate/migrations/000004_create_topology_cache_table.up.sql [hunk 2]
```diff
@@ -0,0 +1,32 @@
+CREATE OR REPLACE FUNCTION set_timestamp() RETURNS TRIGGER LANGUAGE plpgsql AS '
+BEGIN
+  NEW.updated_at = NOW();
+  RETURN NEW;
+END;
+';
```
Comment (danielhochman) [inline backend/cmd/migrate/migrations/000004_create_topology_cache_table.up.sql:6]: can we avoid triggers and do this type of thing server-side?

Comment (mcutalo88) [reply backend/cmd/migrate/migrations/000004_create_topology_cache_table.up.sql:6]: Sure

**Review by danielhochman — APPROVED** [review]

**Pull request merged**